core/blockchain.go

```diff
@@ -1288,16 +1288,6 @@ func (bc *BlockChain) insertBlock(block *types.Block, writes bool) error {
 	blockContentValidationTimer.Inc(time.Since(substart).Milliseconds())
 
 	// No validation errors for the block
-	var activeState *state.StateDB
-	defer func() {
-		// The chain importer is starting and stopping trie prefetchers. If a bad
-		// block or other error is hit however, an early return may not properly
-		// terminate the background threads. This defer ensures that we clean up
-		// and dangling prefetcher, without deferring each and holding on live refs.
-		if activeState != nil {
-			activeState.StopPrefetcher()
-		}
-	}()
```

Comment:
(maybe a silly question)

This is in geth currently:

https://github.com/ethereum/go-ethereum/blob/f4e8e877f6df3fe7d39f07d3c1852c3b43e2451c/core/blockchain.go#L1701-L1710

Why the change? 🤔

Reply:
to call state.WithConcurrentWorkers